packages/launcher/src/main/ipcRequestHandler.js

```diff
@@ -0,0 +1,163 @@
+  },
+  invalidRequest: {
+    message: 'Invalid request',
+    code: 32600,
+  },
+  methodNotFound: {
+    message: 'Method not found',
+    code: 32601,
+  },
+}
+
+let vaultOpen = false
+
+const handleIpcRequests = (
+  mfClient: Client,
+  env: Environment,
+  onLaunchApp: (session: ClientSession) => Promise<any>,
+) => {
+  const vaultConfig = new VaultConfig(env)
+
+  const sdkRequests = {
+    blockchain_getLatestBlock: () => mfClient.blockchain.getLatestBlock(),
+    blockchain_getContractEvents: params =>
+      mfClient.blockchain.getContractEvents(...params),
+    getApiVersion: () => mfClient.apiVersion(),
+  }
+
+  const trustedRequests = {
+    getInstalledApps: (params: Array<any>) =>
+      mfClient.getInstalledApps(...params),
+    removeApp: (params: Array<any>) => mfClient.removeApp(...params),
+    installApp: (params: Array<any>) => mfClient.installApp(...params),
+    createUserIdentity: (params: Array<any>) =>
+      mfClient.createUserIdentity(...params),
+    getOwnUserIdentities: (params: Array<any>) =>
+      mfClient.getOwnUserIdentities(...params),
+
+    launchApp: (params: Array<any>) => launchApp(...params),
+    getVaultsData: () => getVaultsData(),
+    openVault: (params: Array<any>) => openVault(...params),
+    createVault: (params: Array<any>) => createVault(...params),
+  }
+
+  ipcMain.on(sdkRequestChannel, async (event, request) => {
+    const res = await handleRequest(request, sdkRequests)
+    sendIpcResponse(event.sender, sdkResponseChannel, res)
+  })
+
+  ipcMain.on(trustedRequestChannel, async (event, request) => {
+    const res = await handleRequest(request, trustedRequests)
+    sendIpcResponse(event.sender, trustedResponseChannel, res)
+  })
+
+  const handleRequest = async (
+    request: Object,
+    validRequests: Object,
+  ): Promise<ClientResponse> => {
+    if (!request || !request.data || !request.data.method) {
+      return {
+        error: { ...IPC_ERRORS.invalidRequest, method: request.method },
+        id: request.id,
+      }
+    }
+
+    // $FlowFixMe: indexer property
+    if (validRequests[request.data.method]) {
+      const args = request.data.args || []
+      try {
+        // $FlowFixMe: indexer property
+        const res = await validRequests[request.data.method](args)
+        return {
+          id: request.id,
+          result: res,
+        }
+      } catch (err) {
+        return {
+          error: {
+            message: err.message,
+          },
+          id: request.id,
+        }
+      }
+    }
+    return {
+      error: IPC_ERRORS.methodNotFound,
+      id: request.id,
+    }
+  }
+
+  const sendIpcResponse = (sender, channel, response) => {
+    const window = BrowserWindow.fromWebContents(sender)
+    const send = (channel, response) => {
+      if (!(window && window.isDestroyed())) {
+        sender.send(channel, response)
+      }
+    }
+    send(channel, response)
+  }
+
+  const getVaultsData = () => {
+    const vaults = vaultConfig.vaults
```

Comment:
Not an issue but it seems like this could simply be provided as `vaults: vaultConfig.vaults` in the returned object rather than having an extra variable